sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -2661,3 +2661,137 @@ test_suite(
+            # hw/top_earlgrey/data/top_earlgrey.hjson.)
+            echo :::: Set all but one available breakpoints.\\n
+            # Set a breakpoint at the actual address of rom_main(), without
+            # trying to skip the function prologue.
+            break *rom_main
+            break *rom_init
+
+            echo :::: Pretend execution is enabled.\\n
+            set $pc = kRomStartBootExecEn
+
+            echo :::: Continue until rom_main.\\n
+            continue
+            assert_pc rom_main
+
+            echo :::: Continue until rom_init.\\n
+            continue
+            assert_pc rom_init
+
+            echo :::: Finish rom_init.\\n
+            finish
+
+            echo :::: Read and write GPRs.\\n
+            {gpr_copy_commands}
+            {gpr_set_commands}
+            {gpr_restore_commands}
+
+            echo :::: Read selected CSRs.\\n
+            {csr_copy_commands}
+            {csr_set_commands}
+            {csr_restore_commands}
+
+            echo :::: Read and write memory (both SRAM and device).\\n
+            # See hw/top_earlgrey/sw/autogen/top_earlgrey_memory.h
+            set $sram_base = 0x10000000
+            set $sram_len = 0x00008000
+            x/16xb $sram_base
+            x/16xb $sram_base + ($sram_len / 2)
+            x/16xb $sram_base + $sram_len - 16
+
+            echo :::: Manually write bytes to UART0.\\n
+            set $uart0_base = 0x40000000
+            set $uart_wdata = 0x1c
+            print *(uint32_t*)($uart0_base + $uart_wdata) = 'O'
+            print *(uint32_t*)($uart0_base + $uart_wdata) = 'K'
+            print *(uint32_t*)($uart0_base + $uart_wdata) = '!'
+
+            echo :::: Execute code from GDB with `call` command.\\n
+            call uart_putchar('G')
+            call uart_putchar('D')
+            call uart_putchar('B')
+            call uart_putchar('-')
+            call uart_putchar('O')
+            call uart_putchar('K')
+            call uart_putchar('\\r')
+            call uart_putchar('\\n')
+
+            echo :::: GDB script complete.\\n
+            continue
+        """.format(
+            csr_copy_commands = gdb_commands_copy_registers(get_gdb_readable_csr_names()),
+            csr_restore_commands = gdb_commands_restore_registers(get_gdb_settable_csr_names()),
+            csr_set_commands = gdb_commands_set_registers(
+                "0x89abcdef",
+                get_gdb_settable_csr_names(),
+            ),
+            gpr_copy_commands = gdb_commands_copy_registers(IBEX_GPRS),
+            gpr_restore_commands = gdb_commands_restore_registers(IBEX_GPRS),
+            gpr_set_commands = gdb_commands_set_registers("0x89abcdef", IBEX_GPRS),
+        ),
+        gdb_script_symlinks = {
+            "//sw/device/silicon_creator/rom:rom_fpga_cw310.elf": "rom.elf",
+        },
+        rom_bitstream = ":rom_otp_{}_exec_disabled".format(otp_name),
+        rom_kind = "Rom",
+        tags = [
+            "cw310_rom",
```

Comment:
Nit, if you're rerunning CI: can you add the cw310 tag here and also jtag, maybe?

Reply:
They're still there, they're just implicit :) I recently made [opentitan_gdb_fpga_cw310_test](https://cs.opensource.google/opentitan/opentitan/+/master:rules/opentitan_gdb_test.bzl;l=131-134;drc=8b6746068bfd43738de1a2d9d73b08fe9adf4945) into a macro that adds those tags.

Also, to be explicit, I'm intentionally omitting `skip_in_ci`. I'd like to run this test in CI because it's the most bang-for-our-buck GDB test so far.